Sources/WalletConnectSign/Services/HistoryService.swift

```diff
@@ -17,15 +27,33 @@ final class HistoryService {
     func getPendingRequests(topic: String) -> [Request] {
         return getPendingRequests().filter { $0.topic == topic }
     }
-
-    public func getSessionRequest(id: RPCID) -> Request? {
-        guard let record = history.get(recordId: id) else { return nil }
-        return mapRequestRecord(record)
+    
+    func getPendingProposals() -> [Session.Proposal] {
+        let pendingHistory = history.getPending()
+        
+        let requestSubscriptionPayloads = pendingHistory
+            .compactMap { record -> RequestSubscriptionPayload<SessionType.ProposeParams>? in
+                guard let proposalParams = mapProposeParams(record) else {
+                    return nil
+                }
+                return RequestSubscriptionPayload(id: record.id, topic: record.topic, request: proposalParams, publishedAt: Date())
```

Comment:
I don't like we are recreating RequestSubscriptionPayload type here.
could we change type stored by proposalPayloadsStore?